[pt][quant] avgpool3d for graph mode

dskhudia opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36598 [pt][quant] avgpool3d for graph mode**

avgpool3d op for graph mode quantization

Differential Revision: [D21023035](https://our.internmc.facebook.com/intern/diff/D21023035/)

dskhudia committed:
[pt][quant] avgpool3d for graph mode

avgpool3d op for graph mode quantization

Differential Revision: [D21023035](https://our.internmc.facebook.com/intern/diff/D21023035/)

[ghstack-poisoned]

dskhudia committed:
Update on "[pt][quant] avgpool3d for graph mode"


avgpool3d op for graph mode quantization

Differential Revision: [D21023035](https://our.internmc.facebook.com/intern/diff/D21023035/)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1398,7 +1398,9 @@ class M(torch.nn.Module):
+                self.adaptive_avgpool2d = torch.nn.AdaptiveAvgPool2d((1, 1))
+                self.adaptive_avgpool3d = torch.nn.AdaptiveAvgPool3d((1, 1, 1))
                 self.dropout = torch.nn.Dropout()
                 self.avgpool = torch.nn.AvgPool2d(3)
```
looks good, please add `avgpool1d` module as well

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1398,7 +1398,9 @@ class M(torch.nn.Module):
             def __init__(self):
                 super(M, self).__init__()
                 self.maxpool = torch.nn.MaxPool2d(kernel_size=3)
```
and maxpool1d and maxpool3d module

jerryzh168 commented:
basically make sure we have {avg_pool, max_pool, adapative_avg_pool} * {1d, 2d, 3d} * {module, functional}

jerryzh168 commented:
looks like functional maxpool is internal, so we don't need that. and there will be 3 * 3 * 2 - 3 * 3 = 9 different entries for this

dskhudia committed:
Update on "[pt][quant] avgpool3d for graph mode"


avgpool3d op for graph mode quantization

Differential Revision: [D21023035](https://our.internmc.facebook.com/intern/diff/D21023035/)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -1397,18 +1397,36 @@ def test_swap_dequantize_all_ops(self):
                 super(M, self).__init__()
-                self.maxpool = torch.nn.MaxPool2d(kernel_size=3)
-                self.adaptive_avgpool = torch.nn.AdaptiveAvgPool2d((1, 1))
+                self.maxpool1d = torch.nn.MaxPool2d(kernel_size=3)
+                self.maxpool2d = torch.nn.MaxPool2d(kernel_size=3)
+                self.maxpool3d = torch.nn.MaxPool2d(kernel_size=3)
```
MaxPool1d and MaxPool3d are missing

dskhudia replied:
Sorry. Typo.

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -69,8 +69,14 @@ std::vector<std::string> _quantizable_aten_funcs = {
+    "avg_pool1d",
+    "avg_pool2d",
+    "avg_pool3d",
     "_max_pool2d",
```
do we neeed `_max_pool1d` here?

dskhudia committed:
Update on "[pt][quant] avgpool3d for graph mode"


avgpool3d op for graph mode quantization

Differential Revision: [D21023035](https://our.internmc.facebook.com/intern/diff/D21023035/)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
Looks good, thanks!
